numerics/fast_fourier_transform.hpp

```diff
@@ -21,46 +21,59 @@ using base::FloorLog2;
 using geometry::Complexification;
 using geometry::Hilbert;
 using geometry::Interval;
-using quantities::AngularFrequency;
-using quantities::Time;
-
-// This class computes Fourier[{...}, FourierParameters -> {1, -1}] in
-// Mathematica notation (the "signal processing" Fourier transform).
-template<typename Value, std::size_t size_>
+using quantities::Angle;
+using quantities::Derivative;
+using quantities::Difference;
+
+// Given (u₀, ..., uₙ), this class computes the discrete Fourier transform
+//   Uₛ = ∑ᵣ uᵣ exp(-2πirs/n),
+// corresponding to
+//   Fourier[{...}, FourierParameters -> {1, -1}]
+// in Mathematica notation (the "signal processing" Fourier transform).
+template<typename Value, typename Argument, std::size_t size_>
 class FastFourierTransform {
  public:
+  using AngularFrequency = Derivative<Angle, Argument>;
+
   // The size must be a power of 2.
   static constexpr int size = size_;
   static constexpr int log2_size = FloorLog2(size);
   static_assert(size == 1 << log2_size);
 
-  // In the constructors, the container must have |size| elements.  The samples
-  // are assumed to be separated by Δt.
+  // In the constructors, the container must have |size| elements.  For the
+  // purpose of expressing the frequencies, the values are assumed to be
+  // sampled at intervals of Δt.
 
   template<typename Container,
            typename = std::enable_if_t<
                std::is_convertible_v<typename Container::value_type, Value>>>
   FastFourierTransform(Container const& container,
-                       Time const& Δt);
+                       Difference<Argument> const& Δt);
 
   template<typename Iterator,
            typename = std::enable_if_t<std::is_convertible_v<
                typename std::iterator_traits<Iterator>::value_type,
                Value>>>
   FastFourierTransform(Iterator begin, Iterator end,
-                       Time const& Δt);
+                       Difference<Argument> const& Δt);
 
   FastFourierTransform(std::array<Value, size> const& container,
-                       Time const& Δt);
+                       Difference<Argument> const& Δt);
 
   std::map<AngularFrequency, typename Hilbert<Value>::InnerProductType>
   PowerSpectrum() const;
 
   // Returns the interval that contains the largest peak of power.
   Interval<AngularFrequency> Mode() const;
 
+  // Returns the coefficient Uₛ.
+  Complexification<Value> const& operator[](int s) const;
+
+  // Returns the frequency corresponding to Uₛ.
+  AngularFrequency frequency(int s) const;
+
  private:
-  Time const Δt_;
+  Difference<Argument> const Δt_;
```

Comment:
Document that this is only a *bona fide* time if `Argument` is time-like.

Reply:
This is what the type says (and the word time does not appear anyway).